storage/src/checker.rs

```diff
@@ -87,17 +98,93 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
         }
         Ok(())
     }
+
+    /// Wrapper around `find_leaked_areas` that iterates over a collection of ranges.
+    fn find_all_leaked_areas(
+        &self,
+        leaked_ranges: impl IntoIterator<Item = Range<LinearAddress>>,
+    ) -> Result<Vec<(LinearAddress, AreaIndex)>, CheckerError> {
+        leaked_ranges
+            .into_iter()
+            .try_fold(Vec::new(), |mut leaked, range| {
+                let leaked_areas = self.find_leaked_areas(range)?;
+                leaked.extend(leaked_areas);
+                Ok::<_, CheckerError>(leaked)
+            })
+    }
+
+    /// Find all stored areas within `leaked_range`.
+    /// We assume that all space within `leaked_range` are leaked and the stored areas are contiguous.
+    /// Returns error if the last leaked area extends beyond the end of the range.
+    fn find_leaked_areas(
+        &self,
+        leaked_range: Range<LinearAddress>,
+    ) -> Result<Vec<(LinearAddress, AreaIndex)>, CheckerError> {
+        let mut leaked = Vec::new();
+        let mut current_addr = leaked_range.start;
+        loop {
+            let (area_size_index, area_size) = self.area_index_and_size(current_addr)?;
```

Comment:
This is too strict. Let's assume the freed area contains all zeros and 0xff as examples. For the zero case, you'll be iterating over 16 byte chunks for a very long time. For 0xff, you'll be erroring out and not returning any values.